snuba/settings/__init__.py

```diff
@@ -284,6 +284,14 @@ class RedisClusters(TypedDict):
 # avoid spilling over to the next day.
 OPTIMIZE_JOB_CUTOFF_TIME = timedelta(hours=23)
 OPTIMIZE_QUERY_TIMEOUT = 4 * 60 * 60  # 4 hours
+# sleep time to wait for a merge to complete
+OPTIMIZE_BASE_SLEEP_TIME = 300  # 5 mins
+# merges longer than this will be considered long running
+OPTIMIZE_MERGE_MIN_ELAPSED_CUTTOFF_TIME = 10  # 10 mins
+# merges larger than this will be considered long running and will be waited on
+OPTIMIZE_MERGE_SIZE_CUTOFF = 50_000_000_000  # 50GB
+# max number of running merges to wait on
+OPTIMIZE_MERGE_MAX_CONCURRENT_JOBS = 3
```

Comment:
Please rename this variable to reflect jobs running greater than elapsed time cutoff

Also, the value of this constant should be 1. It is possible that you might have 2 merges of 100GB partitions and they satisfy the criteria of being greater than elapsed time. We should pause in those scenarios

Reply:
The current algo sleeps if there's any merge greater than 50GB . So in that case, we would sleep on a single 100GB merge. The `OPTIMIZE_MERGE_MAX_CONCURRENT_JOBS` check is only applied to long running merges of smaller size. So say if there were 3 running merges with elapsed time > 10 min of size 5 GB each, it would sleep.